bin/copyRawDataToTmp.sh

```diff
@@ -47,21 +47,45 @@ fi
 function rsyncNGSRuns() {
 	local _samplesheet
 	_samplesheet="${1}"
-
+	log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Working on ${_samplesheet}"
 	while read -r line
 	do
 		## line is ${filePrefix/${filePrefix}_1.fq.gz
 		##
-		rsync -rlptDvc --relative --rsync-path="sudo -u ${group}-ateambot rsync" "${WORKING_DIR}/./rawdata/ngs/${line}"* "${DESTINATION_DIAGNOSTICS_CLUSTER}:${TMP_ROOT_DIR}/" \
-		|| {
-			log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "Failed to rsync ${line}"
-			log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "    from ${WORKING_DIR}/./rawdata/ngs/"
-			log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "    to ${DESTINATION_DIAGNOSTICS_CLUSTER}:${TMP_ROOT_DIR}/rawdata/ngs/"
+		log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Looping through all prm mounts"
+		copied="no"
+		for prm in "${ALL_PRM[@]}"
+		do
+			log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Checking for ${line} on ${prm} and if it still needs to be processed"
+			if [[ -f "/groups/${group}/${prm}/rawdata/ngs/${line}" && "${copied}" == "no" ]]
+			then	
+				log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "${line} found on ${prm}, start rsyncing.."
+				rsync -rlDvc --relative --rsync-path="sudo -u ${group}-ateambot rsync" "/groups/${group}/${prm}/./rawdata/ngs/${line}"* "${DESTINATION_DIAGNOSTICS_CLUSTER}:${TMP_ROOT_DIR}/" \
+				|| {
+				log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "Failed to rsync ${line}"
+				log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "    from /groups/${group}/${prm}/rawdata/ngs/"
+				log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "    to ${DESTINATION_DIAGNOSTICS_CLUSTER}:${TMP_ROOT_DIR}/rawdata/ngs/"
+				mv "${JOB_CONTROLE_FILE_BASE}."{started,failed}
+				return
+				}
+				copied="yes"
+			else
+				log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "${line} not found on ${prm} OR data has been copied already and we can skip the step"
+			fi	
+		done
+		## if data is still not being copied it is apparently not on prm
+		if [[ "${copied}" == "no" ]]
+		then
+			log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "Rsync failed, data is not found! Searched for the following:"
+			for prm in "${ALL_PRM[@]}"
+			do
+				log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "/groups/${group}/${prm}/rawdata/ngs/${line}"
+			done
 			mv "${JOB_CONTROLE_FILE_BASE}."{started,failed}
-			return
-		}
+			exit 1
```

Comment:
`exit 1` betekent dat het script eruit klapt. Aangezien deze functie wordt aangeroepen in een loop ```for sampleSheet in "${sampleSheetsFolder[@]}"``` worden eventuele volgende samplesheets niet meer verwerkt. Kortom een fout voor één samplesheet, maakt dat de verwerking van de rest stokt. Dat lijkt me niet de bedoeling. Er moet een "*.failed" gemaakt worden om de signaleren via `notifications.sh` dat het mis is gegaan met deze samplesheet, maar daarna moet dit script wel doorgaan en de rest verwerken.